packages/schema/README.md

```diff
@@ -2785,6 +2785,37 @@ import * as S from "@effect/schema/Schema";
 const Email = S.pattern(/^(?!\.)(?!.*\.\.)([A-Z0-9_+-\.]*)[A-Z0-9_+-]@([A-Z0-9][A-Z0-9\-]*\.)+[A-Z]{2,}$/i);
 ```
 
+## Url
+
+Multiple environments like the Browser or Node provide a built-in `URL` class that can be used to validate URLs. Here we demonstrate how to leverage it to validate if a string is a valid URL while the returned value is still a string.
+
+```ts
+import * as S from "@effect/schema/Schema";
+import * as ParseResult from "@effect/schema/ParseResult";
+
+const Url: S.Schema<string, string> = S.transformOrFail(
+  S.string,
+  S.string,
```

Comment:
this should be a schema representing a valid URL, `S.string` represents *any* string

Reply:
This was intentional. To clarify my intend: In the past 1,5 years working with Zod for example I never needed a URL instance. I only used Zod to validate it's a valid URL to then store the data as a string in SQL db as a string.

I see several options:
1. keep the string
2. keep the string and add a second example returning a URL instance
3. only provide the URL example
4. something else you haven in mind?

Let me know what you think makes the most sense

Reply:
Sorry, I wasn't clear. Above, I meant to say a schema representing a valid URL **string** (not in the sense of using the `URL` constructor), which is nothing more than what you are about to define using `filter`

Reply:
To recap: the example could include only the definition of the filter (`UrlString` below), but if you want to include a transformation, it must use the filter as the target schema, something like:

```ts
import * as S from "@effect/schema/Schema"
import * as ParseResult from "@effect/schema/ParseResult"

// a string filter, should be enough
const UrlString: S.Schema<string, string> = S.string.pipe(
  S.filter((value) => {
    try {
      new URL(value)
      return true
    } catch (_) {
      return false
    }
  })
)

// a string transformation
const NormalizeUrl: S.Schema<string, string> = S.transformOrFail(
  S.string,
  UrlString,
  (value, _, ast) =>
    ParseResult.try({
      try: () => new URL(value).toString(),
      catch: (err) =>
        ParseResult.parseError([ParseResult.type(ast, value, err instanceof Error ? err.message : undefined)])
    }),
  ParseResult.succeed
)
```

Reply:
@gcanti I suggested the `transformOrFail` approach to also normalize the URLs (e.g. trailing forward slash). I tend to prefer normalization in schemas to reduce possible permutations of the same value e.g. for caching purposes (where applicable). 

I can see however, how `filter` might be a simpler showcase here ...

Reply:
mmh I think I got the filter wrong, I guess should be

```ts
import * as S from "@effect/schema/Schema"
import * as ParseResult from "@effect/schema/ParseResult"

const NormalizedUrlString: S.Schema<string, string> = S.string.pipe(
  S.filter((value) => {
    try {
      return new URL(value).toString() === value
    } catch (_) {
      return false
    }
  })
)

const NormalizeUrlString: S.Schema<string, string> = S.transformOrFail(
  S.string,
  NormalizedUrlString,
  (value, _, ast) =>
    ParseResult.try({
      try: () => new URL(value).toString(),
      catch: (err) =>
        ParseResult.parseError([ParseResult.type(ast, value, err instanceof Error ? err.message : undefined)])
    }),
  ParseResult.succeed
)
```

Reply:
Thanks a lot! Helps to clarify a lot how to use effect/schema

I pushed a new version which has two examples:
1. one only doing simple validation
2. your normalization example using `transformOrFail`

I tested both on https://stackblitz.com/edit/effect-ts-examples-q6pwrk?file=package.json,src%2Findex.ts